src/transformers/models/clap/feature_extraction_clap.py

```diff
@@ -0,0 +1,350 @@
+            sampling_rate=sampling_rate,
+            padding_value=padding_value,
+            return_attention_mask=return_attention_mask,
+            **kwargs,
+        )
+        self.top_db = top_db
+        self.truncation = truncation
+        self.padding = padding
+        self.n_fft = n_fft
+        self.hop_length = hop_length
+        self.chunk_length_s = chunk_length_s
+        self.nb_max_samples = chunk_length_s * sampling_rate
+        self.sampling_rate = sampling_rate
+        self.frequency_min = frequency_min
+        self.frequency_max = frequency_max
+        self.mel_filters = self.get_mel_filter_banks(
+            n_freqs=int(1 + n_fft // 2),
+            n_mels=feature_size,
+            frequency_min=frequency_min,
+            frequency_max=frequency_max,
+            sample_rate=sampling_rate,
+            norm=None,
+            mel_scale="htk",
+        )
+        self.mel_filters_slaney = self.get_mel_filter_banks(
+            n_freqs=int(1 + n_fft // 2),
+            n_mels=feature_size,
+            frequency_min=frequency_min,
+            frequency_max=frequency_max,
+            sample_rate=sampling_rate,
+            norm="slaney",
+            mel_scale="slaney",
+        )
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes this instance to a Python dictionary.
+
+        Returns:
+            `Dict[str, Any]`: Dictionary of all the attributes that make up this configuration instance, excpet for the
+            mel filter banks, which do not need to be saved or printed as they are too long.
+        """
+        output = copy.deepcopy(self.__dict__)
+        output["feature_extractor_type"] = self.__class__.__name__
+        if "mel_filters" in output:
+            del output["mel_filters"]
+        if "mel_filters_slaney" in output:
+            del output["mel_filters_slaney"]
+        return output
+
+    def _np_extract_fbank_features(self, waveform: np.array, mel_filters: Optional[np.array] = None) -> np.ndarray:
+        """
+        Compute the log-Mel spectrogram of the provided `waveform` using the `hanning` window. In CLAP, two different
+        banks of filters are used depending on the truncation pattern:
+            - `self.mel_filters`: they correspond to the defaults parameters of `torchaduio` which can be obtained from
+              calling `torchaudio.transforms.MelSpectrogram().mel_scale.fb`. These filters are used when `truncation`
+              is set to `fuison`.
+            - `self.mel_filteres_slaney` : they correspond to the defaults parameters of `torchlibrosa` which used
+              `librosa.filters.mel` when computing the mel spectrogram. These filters were only used in the original
+              implementation when the truncation mode is not `"fusion"`.
+        """
+        window = np.hanning(self.n_fft + 1)[:-1]
+        frames = self._fram_wave(waveform)
+        stft = self._stft(frames, window=window)
```

Comment:
Good with me to go in `audio_utils`! Think this is the best location for them. I wouldn't want to put them under `SequenceFeatureExtractor` since not all feature extractors use them, but keeping them universal makes sense if they can be re-used 1-for-1 each time